packages/eslint-plugin/tests/rules/naming-convention.test.ts

```diff
@@ -637,6 +637,22 @@ ruleTester.run('naming-convention', rule, {
         },
       ],
     },
+    {
+      code: `
+        const CONSTANT_VALUE = 42;
+        function inner() {
+          const localConstant = 2;
+        }
+      `,
+      parserOptions,
```

Comment:
nit - you don't need to use the `parserOptions` here, and instead should do:
```suggestion
      parserOptions: {
        sourceType: 'script',
      },
```

Reply:
Additionally, please duplicate this test case and additionally use `sourceType: 'module'` to ensure both scripts and modules work as expected.